vscode/src/telemetry.ts

```diff
@@ -0,0 +1,77 @@
+import * as vscode from "vscode";
+import TelemetryReporter, {
+  TelemetryEventMeasurements,
+  TelemetryEventProperties,
+} from "@vscode/extension-telemetry";
+import { log } from "qsharp-lang";
+
+export enum EventType {
+  DebugSessionStart = "DebugSessionStart",
+  Initialize = "Initialize",
+  LoadLanguageService = "LoadLanguageService",
+  QSharpJupyterCellInitialized = "QSharpJupyterCellInitialized",
+}
+
+type Empty = { [K in any]: never };
+
+type EventTypes = {
+  [EventType.Initialize]: {
+    properties: Empty;
+    measurements: Empty;
+  };
+  [EventType.DebugSessionStart]: {
+    properties: Empty;
+    measurements: {
+      timeToStartMs: number;
+    };
+  };
+  [EventType.LoadLanguageService]: {
+    properties: Empty;
+    measurements: {
+      timeToStartMs: number;
+    };
+  };
+  [EventType.QSharpJupyterCellInitialized]: {
+    properties: Empty;
+    measurements: Empty;
+  };
+};
+
+type WrappedTelemetryEvent = {
+  id: string;
+  data?: {
+    measurements: TelemetryEventMeasurements;
+    properties: TelemetryEventProperties;
+  };
+};
+
+export function initTelemetry() {
+  const extension = vscode.extensions.getExtension("quantum.qsharp-vscode");
```

Comment:
Pass the context in from the extension activation and get the current extension from the context (see https://code.visualstudio.com/api/references/vscode-api#ExtensionContext). Especially as we'll have two different extensions (stable and 'dev'), we want to avoid hardcoding the id in anywhere.